langworld_db_data/adders/listed_value_adder.py

```diff
@@ -89,6 +131,57 @@ def _add_to_inventory_of_listed_values(
 
         return id_of_new_value
 
+    @staticmethod
+    def _get_indices_and_their_line_numbers_in_features_listed_values(
+        rows: list[dict[str, str]],
+        feature_id: str,
+        new_value_en: str,
+        new_value_ru: str,
+    ) -> list[dict[str, int]]:
```

Comment:
I would suggest returning a tuple, not a list. It should make sure that no sequences are unpredictably mutated. We should stick to tuples unless we're absolutely sure we need the mutation.

Within the function, it has to be a list, obviously. But outside of it, it should be a non-mutable sequence of dictionaries, because we should be only iterating over it (or copying it if necessary), not changing it directly.

Reply:
Done! However, I get a warning:
```
Expected type 'tuple[dict[str, int]]', got 'tuple[dict[str, int], ...]' instead 
```

Reply:
Yes, that is often confusing. The ellipsis `...` is needed to indicate that `tuple` can have multiple members. I know it's different from what type hint for `list` looks like, but there are reasons for it.

So just copy this thing from `got... instead` and you'll be fine :)

Reply:
Updated all the typehints around this problem (however not in a separate commit, those changes are united with  renaming `_get_indices_and_their_line_numbers_in_features_listed_values`)